src/cmake/cmakeListsBuilder.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Espressif Systems (Shanghai) CO LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { EOL } from "os";
+import { join } from "path";
+import { Uri } from "vscode";
+import { CmakeListsElement } from "./cmakeListsElement";
+import { pathExists, readFile, readJSON, writeFile } from "fs-extra";
+
+export enum CMakeListsType {
+  Component = "component",
+  Project = "project",
+}
+
+export async function loadCmakeListBuilder(
+  extensionPath: string,
+  type: CMakeListsType
+) {
+  const cmakeListsSchemaFile = join(
+    extensionPath,
+    "src",
+    "cmake",
+    "cmakeListsSchema.json"
+  );
+  const doesSchemaExists = await pathExists(cmakeListsSchemaFile);
+  if (!doesSchemaExists) {
+    return;
+  }
+  const schemaJson = await readJSON(cmakeListsSchemaFile);
+
+  // Component and Project settings are loaded together
+  // Should refactor if there is a way to identify component/project CmakeLists.txt
+  console.log(schemaJson[type]);
+  return schemaJson[type] as CmakeListsElement[];
+  return [].concat(
+    schemaJson.project as CmakeListsElement[],
+    schemaJson.component as CmakeListsElement[]
+  ) as CmakeListsElement[];
+}
+
+export async function updateWithValuesCMakeLists(
+  fileUri: Uri,
+  list: CmakeListsElement[]
+) {
+  const fileString = await readFile(fileUri.fsPath, "utf-8");
+  return parseCmakeListsText(fileString, list);
+}
+
+export function parseCmakeListsText(
+  cmakeListFileText: string,
+  cmakeLists: CmakeListsElement[]
+) {
+  for (const element of cmakeLists) {
+    if (element.type === "array") {
+      const regex = new RegExp(element.regex);
+      const resultStr = cmakeListFileText.match(regex);
+      if (resultStr && resultStr[1].trim().split(" ").length > 0) {
+        element.value = resultStr[1].trim().replace(/\"/g, "").split(" ");
+      }
+    } else {
+      const regex = new RegExp(element.regex);
+      const resultStr = cmakeListFileText.match(regex);
+      if (resultStr && resultStr[1].trim().length > 0) {
+        element.value = [resultStr[1].trim()];
+      }
+    }
+  }
+  return cmakeLists;
+}
+
+export async function updateCmakeListFile(
```

Comment:
One thing I notice, is that this overrides any comment user wrote, maybe we can try to persist comment, else can provide this caution in docs as well as popup for first time